src/Symfony/Component/Cache/Adapter/AbstractTagAwareAdapter.php

```diff
@@ -149,6 +167,17 @@ static function ($deferred, &$expiredIds) use ($getId, $tagPrefix) {
      */
     abstract protected function doInvalidate(array $tagIds): bool;
 
+    protected function doFetchTags(array $ids): iterable
```

Comment:
This is a hook for future improvements / adapters that know how to store values and their tags separately.

Reply:
Ok 👍 

For Redis it makes sense, for filesystem as long as there is no way to read several files in parallel / async, it's going to penalize read performance.

Reply:
Yep, that's why it's implementation specific, each adapter decides / can implement.